### ISSUE TITLE
Hive engine: reth fails to reach TTD on `WaitForTTD`

### ISSUE THREAD

**Issue by Rjected** [paradigmxyz/reth#2689]
### Describe the bug

For example, in the hive test `Pre-TTD ForkchoiceUpdated After PoS Switch (reth)`, we fail with `CLMocker: Error while waiting for TTD: context deadline exceeded`. This is thrown in the hive method `WaitForTTD`, which is used in many hive tests:
https://github.com/ethereum/hive/blob/bbd52d324914d737469e666d5e5279b4ed01f8c6/simulators/ethereum/engine/suites/engine/tests.go#L1497-L1500

### Steps to reproduce

Run hive: `./hive --client reth --sim "ethereum/engine" --sim.limit "engine-api/Pre-TTD ForkchoiceUpdated After PoS Switch"`

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by Rjected** [comment]
There are multiple things causing this:
 * the base fee is not set correctly in auto seal consensus
 * `reth node` uses the network client to initialize the consensus engine, causing single full block downloads to fail instead of use the clients provided by auto sealing

**Comment by Rjected** [comment]
After the linked PRs, this is the error:
```
>> (e272f899) {"jsonrpc":"2.0","id":14,"method":"eth_getBlockByNumber","params":["0x7",false]}
<< (e272f899) {"jsonrpc":"2.0","result":{"hash":"0x78cbf08311ebc263899b869c923754174c436aa2a6c3201b79fc161064e45bad","parentHash":"0xaaf84764e57ad0f363190066ffcfdf8bfb78bc21a5d3e1830dba3a2e5dd2f741","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","miner":"0x0000000000000000000000000000000000000000","stateRoot":"0xdfd39d6af1ced0a198014dddf2543f031c575145cf556d83f5338f6ba157c78b","transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","difficulty":"0x0","number":"0x7","gasLimit":"0x1c9c380","gasUsed":"0x0","timestamp":"0x64630d92","extraData":"0x8b726574682f76302e312e30","mixHash":"0x415892a43098f48a70afd0d5610bc3b2ae335c48cd5025e158aa8e99edd1350b","nonce":"0x0000000000000000","baseFeePerGas":"0x17681061","totalDifficulty":"0x30002","uncles":[],"transactions":[],"size":"0x20e"},"id":14}
>> (e272f899) {"jsonrpc":"2.0","id":21,"method":"engine_forkchoiceUpdatedV1","params":[{"headBlockHash":"0x67ead97eb79b47a1638659942384143f36ed44275d4182799875ab5a87324055","safeBlockHash":"0x67ead97eb79b47a1638659942384143f36ed44275d4182799875ab5a87324055","finalizedBlockHash":"0x67ead97eb79b47a1638659942384143f36ed44275d4182799875ab5a87324055"},null]}
<< (e272f899) {"jsonrpc":"2.0","result":{"payloadStatus":{"status":"SYNCING","latestValidHash":null,"validationError":null},"payloadId":null},"id":21}
DEBUG (Pre-TTD ForkchoiceUpdated After PoS Switch): Failed `Expect*` routine called from: file=/source/suites/engine/tests.go, line=1513
FAIL (Pre-TTD ForkchoiceUpdated After PoS Switch): Unexpected status response on EngineForkchoiceUpdatedV1: SYNCING, expected=INVALID
CLMocker: Closing engine client e272f899
CLMocker: Closed engine client e272f899
End test (reth): Pre-TTD ForkchoiceUpdated After PoS Switch
```

`0x67ead97eb79b47a1638659942384143f36ed44275d4182799875ab5a87324055` is the genesis hash, which we should detect is "pre-merge" and return `INVALID` on the final `ForkchoiceUpdated` call.
